### PR TITLE
fix(ivy): fix property names of ngOnChanges

### PR DESCRIPTION
**Pull request by mhevery**
- #reslove FW-812
- #reslove FW-844

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/core/src/render3/definition.ts [hunk 1]
```diff
@@ -345,35 +345,44 @@ export function defineNgModule<T>(def: {type: T} & Partial<NgModuleDef<T>>): nev
  *   @Input()
  *   propName1: string;
  *
- *   @Input('publicName')
- *   propName2: number;
+ *   @Input('publicName2')
+ *   daclaredPropName2: number;
```
Comment (trotyl) [inline packages/core/src/render3/definition.ts]: daclared -> declared

Comment (mhevery) [reply packages/core/src/render3/definition.ts]: fixed

**Comment by mhevery** [comment]
http://test/OCL:225939484:BASE:225939526:1545111699258:7e00764e

File: packages/upgrade/test/static/integration/upgrade_component_spec.ts [hunk 2]
```diff
@@ -465,240 +465,252 @@ withEachNg1Version(() => {
     });
 
     describe('bindings', () => {
-      it('should support `@` bindings', fakeAsync(() => {
-           let ng2ComponentInstance: Ng2Component;
+      fixmeIvy('FW-724: upgraded ng1 components are not being rendered')
```
Comment (petebacondarwin) [inline packages/upgrade/test/static/integration/upgrade_component_spec.ts]: This is not actually failing.

File: packages/upgrade/test/static/integration/upgrade_component_spec.ts [hunk 3]
```diff
@@ -465,240 +465,252 @@ withEachNg1Version(() => {
...
-           });
-         }));
+                  expect(multiTrim(element.textContent))
+                      .toBe('Inside: foo2, bar2 | Outside: foo2, bar2');
+                });
+              }));
 
-      it('should support `<` bindings', fakeAsync(() => {
-           let ng2ComponentInstance: Ng2Component;
+      fixmeIvy('FW-724: upgraded ng1 components are not being rendered')
```
Comment (petebacondarwin) [inline packages/upgrade/test/static/integration/upgrade_component_spec.ts]: Nor this.

**Review by petebacondarwin — APPROVED** [review]
I added a fixup commit that re-enables the tests that were actually passing.
Otherwise this looks good to me, although I would have put that are just renaming `declared...` and `public...` usage in the code in its own commit. Otherwise it confuses what the main commit is trying to do.

**Pull request closed without merging**